### DISCUSSION TITLE
Why modules' build.gradle file can work with only namespace declaration?

### DISCUSSION THREAD

**Discussion by khahani** [android/nowinandroid#946] [General]
How can I make such a concise build.gradle file for modules:

```
android {
    namespace = "com.google.samples.apps.nowinandroid.core.common"
}
```

Normally I should do this for each module: 

```
android {
    namespace 'androidx.compose.samples.common'
    compileSdk 33

    defaultConfig {
        minSdk 24
    }
    
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_17
        targetCompatibility JavaVersion.VERSION_17
    }
    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_17
    }
}
```

**Comment by SimonMarquis** [comment]
You can extract all the common configurations as a [Gradle convention plugin](https://docs.gradle.org/current/userguide/sharing_build_logic_between_subprojects.html#sec:convention_plugins).
This is what is used in this project: the conventions plugins are located in [build-logic/convention](https://github.com/android/nowinandroid/tree/main/build-logic/convention/src/main/kotlin).

For example:
The [AndroidLibraryConventionPlugin.kt](https://github.com/android/nowinandroid/blob/main/build-logic/convention/src/main/kotlin/AndroidLibraryConventionPlugin.kt) is registered as `nowinandroid.android.library` here:

https://github.com/android/nowinandroid/blob/151f877bbe6546c78f8f03f16e56564e285fb508/build-logic/convention/build.gradle.kts#L63-L66

and used like this in an android library module:

https://github.com/android/nowinandroid/blob/151f877bbe6546c78f8f03f16e56564e285fb508/core/common/build.gradle.kts#L16-L20
